lambda/lambda.ml

```diff
@@ -297,7 +297,8 @@ type lambda =
   | Lfunction of lfunction
   | Llet of let_kind * value_kind * Ident.t * lambda * lambda
   | Lmutlet of value_kind * Ident.t * lambda * lambda
-  | Lletrec of (Ident.t * lambda) list * lambda
+  | Lletrec of
+      (Ident.t * Typedtree.recursive_binding_kind * lambda) list * lambda
```

Comment:
I would be in favor of making this a record: you are already touching all those places so this will not make your patch more invasive, and it is likely to help in the future.

Reply:
I will try this suggestion. I think the only place where it could cause problems is is `tmc.ml`, but hopefully my GADT hack will prove robust enough to handle that.